lib/mp4/probe.js

```diff
@@ -87,52 +88,55 @@ timescale = function(init) {
  * fragment, in seconds
  */
 startTime = function(timescale, fragment) {
-  var trafs, baseTimes, result;
+  var trafs, result;
 
   // we need info from two childrend of each track fragment box
   trafs = findBox(fragment, ['moof', 'traf']);
 
   // determine the start times for each track
-  baseTimes = [].concat.apply([], trafs.map(function(traf) {
-    return findBox(traf, ['tfhd']).map(function(tfhd) {
-      var id, scale, baseTime;
-
-      // get the track id from the tfhd
-      id = toUnsigned(tfhd[4] << 24 |
-                      tfhd[5] << 16 |
-                      tfhd[6] <<  8 |
-                      tfhd[7]);
-      // assume a 90kHz clock if no timescale was specified
-      scale = timescale[id] || 90e3;
-
-      // get the base media decode time from the tfdt
-      baseTime = findBox(traf, ['tfdt']).map(function(tfdt) {
-        var version, result;
-
-        version = tfdt[0];
-        result = toUnsigned(tfdt[4] << 24 |
-                            tfdt[5] << 16 |
-                            tfdt[6] <<  8 |
-                            tfdt[7]);
-        if (version ===  1) {
-          result *= Math.pow(2, 32);
-          result += toUnsigned(tfdt[8]  << 24 |
-                               tfdt[9]  << 16 |
-                               tfdt[10] <<  8 |
-                               tfdt[11]);
-        }
-        return result;
-      })[0];
-      baseTime = typeof baseTime === 'number' && !isNaN(baseTime) ? baseTime : Infinity;
+  var lowestTime = trafs.reduce(function(acc, traf) {
+    var tfhd = findBox(traf, ['tfhd'])[0];
+
+    // get the track id from the tfhd
+    var id = toUnsigned(tfhd[4] << 24 |
+      tfhd[5] << 16 |
+      tfhd[6] <<  8 |
+      tfhd[7]);
+    // assume a 90kHz clock if no timescale was specified
+    var scale = timescale[id] || 90e3;
+
+    // get the base media decode time from the tfdt
+    var tfdt = findBox(traf, ['tfdt'])[0];
+    var dv = new DataView(tfdt.buffer, tfdt.byteOffset, tfdt.byteLength);
+    var baseTime;
+
+    // version 1 is 64 bit
+    if (tfdt[0] === 1) {
+      baseTime = getUint64(tfdt.subarray(4, 12));
+    } else {
+      baseTime = dv.getUint32(4);
+    }
 
-      // convert base time to seconds
-      return baseTime / scale;
-    });
-  }));
+    // convert base time to seconds if it is a valid number.
+    let seconds;
+    if (typeof baseTime === 'bigint') {
+      seconds = baseTime / globalThis.BigInt(scale);
+    } else if (typeof baseTime === 'number' && !isNaN(baseTime)) {
+      seconds = baseTime / scale;
+    }
+
+    if (seconds < Number.MAX_SAFE_INTEGER) {
```

Comment:
if converting it to the appropriate scale, brings it below the threshold to be a bigInt bring it back to a number.